frontend/public/components/pod.tsx

```diff
@@ -215,109 +223,110 @@ const podColumnInfo = Object.freeze({
-        id: podColumnInfo.namespace.id,
-        sortField: 'metadata.namespace',
-        transforms: [sortable],
-        props: { className: podColumnInfo.namespace.classes },
-      },
-      {
-        title: i18next.t(podColumnInfo.status.title),
-        id: podColumnInfo.status.id,
-        sortFunc: 'podPhase',
-        transforms: [sortable],
-        props: { className: podColumnInfo.status.classes },
-      },
-      {
-        title: i18next.t(podColumnInfo.ready.title),
-        id: podColumnInfo.ready.id,
-        sortFunc: 'podReadiness',
-        transforms: [sortable],
-        props: { className: podColumnInfo.ready.classes },
-      },
-      {
-        title: i18next.t(podColumnInfo.restarts.title),
-        id: podColumnInfo.restarts.id,
-        sortFunc: 'podRestarts',
-        transforms: [sortable],
-        props: { className: podColumnInfo.restarts.classes },
-      },
-      {
-        title: showNodes
-          ? i18next.t(podColumnInfo.node.title)
-          : i18next.t(podColumnInfo.owner.title),
-        id: podColumnInfo.owner.id,
-        sortField: showNodes ? 'spec.nodeName' : 'metadata.ownerReferences[0].name',
-        transforms: [sortable],
-        props: { className: podColumnInfo.owner.classes },
-      },
-      {
-        title: i18next.t(podColumnInfo.memory.title),
-        id: podColumnInfo.memory.id,
-        sortFunc: 'podMemory',
-        transforms: [sortable],
-        props: { className: podColumnInfo.memory.classes },
-      },
-      {
-        title: i18next.t(podColumnInfo.cpu.title),
-        id: podColumnInfo.cpu.id,
-        sortFunc: 'podCPU',
-        transforms: [sortable],
-        props: { className: podColumnInfo.cpu.classes },
-      },
-      {
-        title: i18next.t(podColumnInfo.created.title),
-        id: podColumnInfo.created.id,
-        sortField: 'metadata.creationTimestamp',
-        transforms: [sortable],
-        props: { className: podColumnInfo.created.classes },
-      },
-      {
-        title: i18next.t(podColumnInfo.node.title),
-        id: podColumnInfo.node.id,
-        sortField: 'spec.nodeName',
-        transforms: [sortable],
-        props: { className: podColumnInfo.node.classes },
-        additional: true,
-      },
-      {
-        title: i18next.t(podColumnInfo.labels.title),
-        id: podColumnInfo.labels.id,
-        sortField: 'metadata.labels',
-        transforms: [sortable],
-        props: { className: podColumnInfo.labels.classes },
-        additional: true,
-      },
-      {
-        title: i18next.t(podColumnInfo.ipaddress.title),
-        id: podColumnInfo.ipaddress.id,
-        sortField: 'status.podIP',
-        transforms: [sortable],
-        props: { className: podColumnInfo.ipaddress.classes },
-        additional: true,
-      },
-      {
-        title: '',
-        props: { className: Kebab.columnClass },
-      },
-    ];
-  };
-};
+const getColumns = (showNodes: boolean, t: TFunction): TableColumn<PodKind>[] => [
```

Comment:
IMO, it's generally better to use `i18next.t` rather than passing it as a parameter. Is there a reason that needed to change? (Not all `t`s are the same since `useTranslation` takes arguments, and it means one less parameter you have to pass.)

Reply:
Im seeing that the `useTranslation` will return a new instance of `t` when the language is changed. And all useMemo-s etc are correctly re-evaluated. And thats exactly what I expect from the `useTranslation` hook otherwise it wouldn't be responsive .. ?

I want to memoize result of `getColumns` otherwise I'd be passing a new instance of array on every re-render.
If i use `i18next.t` in `getColumns` with useMemo then its not responsive.